### ISSUE TITLE
Fx.hover does not work for scattergl

### ISSUE THREAD

**Issue by kyle-strait** [plotly/plotly.js#1963]
Hello,

I recently switched my app from using scatter to scattergl and realized that the Fx.hover method does not work correctly for that plot type. This [codepen](https://codepen.io/kyle-strait/pen/gxvzLX) shows that the change to scattergl breaks the hover call. I did not see this listed anywhere as a gl2d limitation so I wanted to get it out there. I also realize that the plan is for Fx to be removed in the future however I was hoping there may be some plan to amend the hover method to work with scattergl.

A quick look through made it seem like one of the sticking points is that the hover method is looking for a "hoveron" field in the trace which is not present for scattergl. It is then set to an empty string and the hover code moves on without looking for any points to execute the hover on.

Thanks!

**Comment by etpinard** [comment]
That's correct, `Fx.hover` only works for cartesian subplots (and maybe `pie` - I don't remember).

You can try using [`Fx.loneHover`](https://github.com/plotly/plotly.js/blob/ff8ecb1b6a63559a897c2ba744cecdef0c6660a8/src/components/fx/hover.js#L122) the same [way](https://github.com/plotly/plotly.js/blob/ff8ecb1b6a63559a897c2ba744cecdef0c6660a8/src/plots/gl2d/scene2d.js#L681) our gl2d code invokes it to get the desired result.

`Fx.hover` is one our _unofficial_ methods that we'll remove in [`v2`](https://github.com/plotly/plotly.js/issues/420), maybe we should we should consider adding an official `Plotly.hover` method?

**Comment by kyle-strait** [comment]
Hi @etpinard thank you for your reply.

I will give Fx.loneHover a try and see if I can get it to do what we need.

As for an official Plotly.hover method I personally think it could be very helpful. In our case we have points that are being used in multiple locations (i.e. a map and the plots) and they are linked. I am currently using the hover capabilities to perform "highlighting" when there is a hover on a point in a map that links to a point on the plot. I don't know how common of a use case that is but it is pretty key to the application we are working on.

**Comment by lnunno** [comment]
It'd be nice if this was mentioned in the Plotly docs since [the official docs use Fx.hover directly](https://plot.ly/javascript/hover-events/#triggering-hover-events) with no mention of scattergl.

**Comment by arlowhite** [comment]
@etpinard where is the **scattergl** code that creates hovers in compare mode?
(such that the labels don't overlap each other)
Looking at `loneHover`, I don't see how it deals with multiple points.

I noticed there's `Fx.multiHovers`, but it only seems to be used in `parcats.js`, not sure if this is what I want.

All I want to do is achieve the same behavior as compare mode, but across multiple subplots sharing the same x axis. It'd be nice if there was an option for this. e.g. `hovermode: 'across'` on the xaxis

**Comment by philtgun** [comment]
Any updates on this issue? In my case, my goal is to trigger hover event on the scattergl plot for comparison, and `hover()` doesn't work with scattergl, and `loneHover()` is too cumbersome to use at a high level, and there is not much documentation available except for the code. But apparently, that is the only way to do things at the moment?
